pkg/apiserver/controllers/decisions.go

```diff
@@ -92,56 +92,102 @@ func (c *Controller) DeleteDecisions(gctx *gin.Context) {
 
 func (c *Controller) StreamDecision(gctx *gin.Context) {
 	var data []*ent.Decision
-
 	ret := make(map[string][]*models.Decision, 0)
 	ret["new"] = []*models.Decision{}
 	ret["deleted"] = []*models.Decision{}
 
-	// if the blocker just start, return all decisions
-	if _, ok := gctx.Request.URL.Query()["startup"]; ok {
-		data, err := c.DBClient.QueryAllDecisions()
-		if err != nil {
-			log.Errorf("failed querying decisions: %v", err)
-			gctx.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
-			return
-		}
-		ret["new"], err = FormatDecisions(data)
-		gctx.JSON(http.StatusOK, ret)
-		return
-	}
-
 	val := gctx.Request.Header.Get(c.APIKeyHeader)
 	hashedKey := sha512.New()
 	hashedKey.Write([]byte(val))
 	hashStr := fmt.Sprintf("%x", hashedKey.Sum(nil))
-	lastPull, err := c.DBClient.LastBlockerPull(hashStr)
+	blockerInfo, err := c.DBClient.SelectBlocker(hashStr)
 	if err != nil {
-		gctx.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+		if _, ok := err.(*ent.NotFoundError); ok {
+			gctx.JSON(http.StatusForbidden, gin.H{"message": err.Error()})
+		} else {
+			gctx.JSON(http.StatusInternalServerError, gin.H{"message": "not allowed"})
+		}
+		return
 	}
 
-	lastPullTime, err := time.Parse(time.RFC3339, lastPull)
-	if err != nil {
-		gctx.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+	if blockerInfo == nil {
+		gctx.JSON(http.StatusInternalServerError, gin.H{"message": "not allowed"})
+		return
+	}
+
+	// if the blocker just start, return all decisions
+	if val, ok := gctx.Request.URL.Query()["startup"]; ok {
```

Comment:
you might want to check that the key is present in the map to avoid deref ?

Reply:
this is what the code does no ?